support/utils/functions_test.go

```diff
@@ -36,3 +36,90 @@ func TestDedupe(t *testing.T) {
 		})
 	}
 }
+
+func TestToMapStringInterface_SuccessMap(t *testing.T) {
+	success := map[string]interface{}{
+		"test": true,
+	}
+	m, e := ToMapStringInterface(success)
+	assert.Equal(t, success, m)
+	assert.NoError(t, e)
+}
+
+func TestToMapStringInterface_SuccessStruct(t *testing.T) {
+	type NonEmptyStruct struct {
+		Value int `json:"value"`
+	}
+
+	type SuccessStruct struct {
+		True           bool
+		False          bool
+		EmptyString    string
+		NonEmptyString string
+		Int            int
+		Float          float64
+		EmptyStruct    struct{}
+		NonEmptyStruct NonEmptyStruct
+		EmptyInts      []int
+		NonEmptyInts   []int
+	}
+
+	success := SuccessStruct{
+		True:           true,
+		False:          false,
+		EmptyString:    "",
+		NonEmptyString: "hello world",
+		Int:            1,
+		Float:          1.0,
+		EmptyStruct:    struct{}{},
+		NonEmptyStruct: NonEmptyStruct{1},
+		EmptyInts:      []int{},
+		NonEmptyInts:   []int{1, 2},
+	}
+
+	want := map[string]interface{}{
+		"EmptyInts":      []interface{}{},
```

Comment:
the order of this `want` map should be the same as the `success` struct to make it more readable and easier to follow.

Reply:
PS: since it's not intuitive without the underlying context, please add a comment explaining why we "lose the type information" in places such as `EmptyInts` and `NonEmptyStruct` as examples